frontend/src/app/core/services/auth.service.ts

```diff
@@ -61,22 +66,39 @@ export class AuthService {
         localStorage.setItem('accessToken', userIdToken);
     }
 
-    public setUserSection() {
-        this.userService.getUser().subscribe((resp) => {
-            localStorage.setItem('user', JSON.stringify(resp));
-        });
+    setLocalStorage(user: ILocalStorageUser) {
+        localStorage.setItem('user', JSON.stringify(user));
+        this.user.next(user);
+        this.user.complete();
     }
 
-    public getUserSection() {
-        const userSection = localStorage.getItem('user');
+    setUser() {
+        this.userService.getUser().subscribe(
+            (resp) => {
+                const user = {
+                    firstName: resp.firstName,
+                    lastName: resp.lastName,
+                    imagePath: resp.imagePath,
+                };
+
+                this.setLocalStorage(user);
+            },
+            (err: Error) => {
+                this.logout();
+                this.toastr.showError(err.message, 'Error!');
+            },
+        );
+
+        return this.user.asObservable();
+    }
 
-        if (!userSection) {
-            return null;
-        }
+    getUser() {
```

Comment:
lets go with setUser -> loadUser
And instead of getUser, just subscribe to the subject, it will have latest data